ui-tests/tests/widget_notebook_example.test.ts

```diff
@@ -0,0 +1,138 @@
+    await page.notebook.activate(notebook);
+
+    const captures: (Buffer | null)[] = []; // Array to store cell screenshots
+    const cellCount = await page.notebook.getCellCount();
+
+    // Run all cells and capture their screenshots
+    await page.notebook.runCellByCell({
+      onAfterCellRun: async (cellIndex: number) => {
+        const cell = await page.notebook.getCellOutput(cellIndex);
+        if (cell && (await cell.isVisible())) {
+          captures[cellIndex] = await cell.screenshot(); // Save the screenshot by cell index
+        }
+      },
+    });
+
+    await page.notebook.save();
+
+    // Ensure that each cell's screenshot is captured
+    for (let i = 0; i < cellCount; i++) {
+      const image = `widgets-cell-${i}.png`;
+
+      if (captures[i]) {
+        expect.soft(captures[i]).toMatchSnapshot(image); // Compare pre-existing capture
+        continue;
+      }
+    }
+
+    const widgetCellIndex = 3;
+
+    await waitForWidget(page, widgetCellIndex, 'input[type="checkbox"]');
+    await waitForWidget(page, widgetCellIndex, 'button:has-text("Cluster Down")');
+    await waitForWidget(page, widgetCellIndex, 'button:has-text("Cluster Up")');
+
+    await interactWithWidget(page, widgetCellIndex, 'input[type="checkbox"]', async (checkbox) => {
+      await checkbox.click();
+      const isChecked = await checkbox.isChecked();
+      expect(isChecked).toBe(true);
+    });
+
+    await interactWithWidget(page, widgetCellIndex, 'button:has-text("Cluster Down")', async (button) => {
+      await button.click();
+      const clusterDownMessage = await page.waitForSelector('text=No instances found, nothing to be done.', { timeout: 5000 });
+      expect(clusterDownMessage).not.toBeNull();
+    });
+
+    await interactWithWidget(page, widgetCellIndex, 'button:has-text("Cluster Up")', async (button) => {
+      await button.click();
+
+      const successMessage = await page.waitForSelector('text=Ray Cluster: \'raytest\' has successfully been created', { timeout: 10000 });
+      expect(successMessage).not.toBeNull();
+
+      const resourcesMessage = await page.waitForSelector('text=Waiting for requested resources to be set up...');
+      expect(resourcesMessage).not.toBeNull();
+
+      const upAndRunningMessage = await page.waitForSelector('text=Requested cluster is up and running!');
+      expect(upAndRunningMessage).not.toBeNull();
+
+      const dashboardReadyMessage = await page.waitForSelector('text=Dashboard is ready!');
+      expect(dashboardReadyMessage).not.toBeNull();
+    });
+
+    await runLastCell(page, cellCount, '(<CodeFlareClusterStatus.READY: 1>, True)');
+
+    await interactWithWidget(page, widgetCellIndex, 'button:has-text("Cluster Down")', async (button) => {
+      await button.click();
+      const clusterDownMessage = await page.waitForSelector('text=Ray Cluster: \'raytest\' has successfully been deleted', { timeout: 5000 });
+      expect(clusterDownMessage).not.toBeNull();
```

Comment:
We could guarantee that the cluster has been deleted by making use of the `list_all_clusters()` command. We should assert it returns an empty list in this case. WDYT?

Reply:
Isn't checking for cluster.status() enough?